evap/staff/tests.py

```diff
@@ -1081,28 +1085,96 @@ def load_test_data(self):
+        cls.contribution1 = mommy.make(Contribution, contributor=cls.main_user, course=cls.course1)
+        cls.contribution2 = mommy.make(Contribution, contributor=cls.other_user, course=cls.course1)  # this should make the merge fail
+        cls.contribution3 = mommy.make(Contribution, contributor=cls.other_user, course=cls.course2)
 
-        # see https://github.com/fsr-itse/EvaP/issues/705
-        #self.assertEqual(set(user1.delegates.all()), set([delegate1, delegate2]))
-        #self.assertEqual(set(course2.voters.all()), set([user1]))
+    def test_merge_users(self):
+        merged_user, errors = merge_users(self.main_user, self.other_user)  # merge should fail
+        self.assertSequenceEqual(errors, ['contributions', 'courses_participating_in'])
+
+        # assert that nothing has changed
+        self.assertEqual(self.main_user.username, "main_user")
+        self.assertEqual(self.main_user.title, "Dr.")
+        self.assertEqual(self.main_user.first_name, "Main")
+        self.assertEqual(self.main_user.last_name, "")
+        self.assertEqual(self.main_user.email, "")
+        self.assertSequenceEqual(self.main_user.groups.all(), [self.group1])
+        self.assertSequenceEqual(self.main_user.delegates.all(), [self.user1, self.user2])
+        self.assertSequenceEqual(self.main_user.represented_users.all(), [self.user3])
+        self.assertSequenceEqual(self.main_user.cc_users.all(), [self.user1])
+        self.assertSequenceEqual(self.main_user.ccing_users.all(), [])
+        self.assertEqual(self.other_user.username, "other_user")
+        self.assertEqual(self.other_user.title, "")
+        self.assertEqual(self.other_user.first_name, "Other")
+        self.assertEqual(self.other_user.last_name, "User")
+        self.assertEqual(self.other_user.email, "other@test.com")
+        self.assertSequenceEqual(self.other_user.groups.all(), [self.group2])
+        self.assertSequenceEqual(self.other_user.delegates.all(), [self.user3])
+        self.assertSequenceEqual(self.other_user.represented_users.all(), [self.user1])
+        self.assertSequenceEqual(self.other_user.cc_users.all(), [])
+        self.assertSequenceEqual(self.other_user.ccing_users.all(), [self.user1, self.user2])
+        self.assertSequenceEqual(self.course1.participants.all(), [self.main_user, self.other_user])
+        self.assertSequenceEqual(self.course2.participants.all(), [self.main_user])
+        self.assertSequenceEqual(self.course2.voters.all(), [self.main_user])
+        self.assertSequenceEqual(self.course3.participants.all(), [self.other_user])
+        self.assertSequenceEqual(self.course3.voters.all(), [self.other_user])
+
+        # fix data
+        self.course1.participants = [self.main_user]
+        self.contribution2.delete()
+
+        merged_user, errors = merge_users(self.main_user, self.other_user)  # merge should succeed
+
+        self.assertEqual(self.main_user.username, "main_user")
```

Comment:
...that's probably because you pass `main_user` into `merge_users`, and in there `main_user` is modified. so it might actually not be necessary... i'd add it anyways to make clear what should be tested and to be independent of `merge_users`'s implementation.